### DISCUSSION TITLE
Possibility of Multiple directed layout in a graph - where cells are part of both layout

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2263] [Q&A]
### Introduction

I have one of the situation where i need to enforce both  rankDir: "LR" and rankDir:"RL" for cells in same graph.
Is it possible??
layout.DirectedGraph.layout(LRCells, {
    dagre: dagre,
    graphlib: graphlib,
    nodeSep: 50,
    edgeSep: 50,
    rankSep: 100,
    rankDir: "LR",
    //setVertices: true,
    setLabels: true,
  });

  layout.DirectedGraph.layout(RLCells, {
    dagre: dagre,
    graphlib: graphlib,
    nodeSep: 50,
    edgeSep: 50,
    rankSep: 100,
    rankDir: "RL",
    //setVertices: true,
    setLabels: true,
  });

what I am trying to do is,
For some special cases I want some links to follow LR to connect elements and other links to follow RL to connect elements.

Note:
RLCells: Contains Links to follow RL but these links are connecting elements that are part of LR.

Attaching a image for better understanding:
![image](https://github.com/clientIO/joint/assets/22453886/bae05c33-b45d-4973-aeda-c3e547fe42c1)

case1: 
I have elements and links connecting  1->2->3 in LRCells which is in directed rankDir:"LR"
case2: 
To have a layout that i sdepicted in the screenshot  the 3 -> 4 ->1. It wont work with rankDir "LR". which is fine. 

Now, to make 3->4->1 connection that enforce the layout I am trying to put all the links and elements in RLCells and making the rankDir:RL (as shown in the example above)

It is not working as expected. Any suggestion would be appreciable. 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by ROYopedia** [comment]
Would like to hear about any update or any comment you guys have on this?

**Comment by kumilingus** [comment]
In general, the _Directed Graph Layout_ specific question can be asked directly in the [dagre](https://github.com/dagrejs/dagre) repository (the authors of the layout). DGL is just a wrapper.

You should be specific about what you are trying to achieve. Try to mention more than a simple case.
- Do the two groups of cells have only 2 elements in common? Or any number?
- Is it always the case that the first group is above the second group?
- What is the result of your experiment so far? You said it's not as expected. But how close was it to what you wanted to achieve? Can you share some code or at least a screenshot?
